apis/v1alpha1/opentelemetrycollector_types.go

```diff
@@ -120,11 +120,21 @@ type OpenTelemetryTargetAllocator struct {
 	// +optional
 	Enabled bool `json:"enabled,omitempty"`
 
+	// PrometheusCR defines the configuration for Prometheus custom resource retrieval.
+	// +optional
+	PrometheusCR OpenTelemetryTargetAllocatorPrometheusCR `json:"prometheusCR,omitempty"`
+
 	// Image indicates the container image to use for the OpenTelemetry TargetAllocator.
 	// +optional
 	Image string `json:"image,omitempty"`
 }
 
+type OpenTelemetryTargetAllocatorPrometheusCR struct {
+	// Enabled indicates whether to use a Prometheus custom resources as target or not.
```

Comment:
Which p8s CR does it use? Can there be multiple in a namespace?

Reply:
As this is a MVC all instances of `ServiceMonitor` and `PodMonitor` in all namespaces the service account is able to retrieve from the API server